src/test/java/edu/kit/ipd/crowdcontrol/objectservice/payment/PaymentCalculatorTest.java

```diff
@@ -0,0 +1,132 @@
+     * @throws Exception
+     */
+    @Test
+    public void testEstimatePayment() throws Exception {
+
+        WorkerRecord workerOne = new WorkerRecord(0,"","","",0);
+        WorkerRecord workerTwo = new WorkerRecord(1,"","","",0);
+
+        List<AnswerRecord> answerSetWorkerOne = new ArrayList<>();
+        answerSetWorkerOne.add(new AnswerRecord(0,workerOne.getIdWorker(),"",null,0,9,true));
+        answerSetWorkerOne.add(new AnswerRecord(1,workerOne.getIdWorker(),"",null,0,9,true));
+        answerSetWorkerOne.add(new AnswerRecord(2,workerOne.getIdWorker(),"",null,0,9,true));
+        answerSetWorkerOne.add(new AnswerRecord(3,workerOne.getIdWorker(),"",null,0,9,true));
+        answerSetWorkerOne.add(new AnswerRecord(4,workerOne.getIdWorker(),"",null,0,9,true));
+
+        List<AnswerRecord> answerSetWorkerTwo = new ArrayList<>();
+        answerSetWorkerTwo.add(new AnswerRecord(0,workerTwo.getIdWorker(),"",null,0,9,true));
+        answerSetWorkerTwo.add(new AnswerRecord(1,workerTwo.getIdWorker(),"",null,0,9,true));
+
+
+        workerAnswerMap.put(workerOne,answerSetWorkerOne);
+        workerAnswerMap.put(workerTwo,answerSetWorkerTwo);
+
+        Map<Worker,Integer> resultAnswers = calculator.estimatePayment(exp);
+        assertNotNull(resultAnswers);
+        assertEquals(resultAnswers.size(),2);
+
+        List<Worker> sortedResult = sortWorkerMap(resultAnswers);
+
+        assertEquals((int)resultAnswers.get(sortedResult.get(0)), exp.getPaymentBase()+(exp.getPaymentAnswer()*answerSetWorkerOne.size()));
+        assertEquals((int)resultAnswers.get(sortedResult.get(1)), exp.getPaymentBase()+(exp.getPaymentAnswer()*answerSetWorkerTwo.size()));
+
+
+        List<RatingRecord> ratingSetWorkerTwo = new ArrayList<>();
+        ratingSetWorkerTwo.add(new RatingRecord(0,exp.getId(),0,null,6,"",1,7));
+        ratingSetWorkerTwo.add(new RatingRecord(1,exp.getId(),0,null,5,"",1,9));
+
+        workerRatingMap.put(workerTwo,ratingSetWorkerTwo);
+
+
+        Map<Worker,Integer> resultRatings = calculator.estimatePayment(exp);
+        assertNotNull(resultRatings);
+        assertEquals(resultRatings.size(),2);
+        sortedResult = sortWorkerMap(resultAnswers);
+        assertEquals((int)resultRatings.get(sortedResult.get(0)),  exp.getPaymentBase()+(exp.getPaymentAnswer()*answerSetWorkerOne.size()));
+        assertEquals((int)resultRatings.get(sortedResult.get(1)), exp.getPaymentBase()+(exp.getPaymentAnswer()*answerSetWorkerTwo.size())+(exp.getPaymentRating()*ratingSetWorkerTwo.size()));
+
+
+
+
+
+
+
+
+
+
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void testFalsePaymentArgs(){
+        exp = Experiment.newBuilder().setPaymentBase(-1).setPaymentAnswer(4).setPaymentRating(4).build();
+        calculator.estimatePayment(exp);
+
+    }
+
+
+    private List<Worker> sortWorkerMap(Map<Worker,Integer> map){
+        List<Worker> sortedWorkers = new LinkedList<>(map.keySet());
```

Comment:
just a general advice...don't use LinkedLists in java, they only provide benefits when adding to head or tail of huge lists. But then you should have used Queues.